src/services/authServices.js

```diff
@@ -94,27 +95,32 @@ class authService {
      * @returns
      * @author khushbuw
      */
-    static async generatePassword(id, requestParams,user) {
+    static async generatePassword(id, requestParams, userRole) {
         try {
-            if(requestParams.password == '' || requestParams.cpassword == ''){
-                return {error:config.emptyFields}
+            if ((userRole == 'customer') || (userRole == 'Customer')) {
+                return {
+                    error: config.generatePasswordNotAllow
+                }
             }
-            if(requestParams.password !== requestParams.cpassword){
-                return {error: config.matchPassword}
+            if (requestParams.password == '' || requestParams.cpassword == '') {
+                return { error: config.emptyFields }
+            }
+            if (requestParams.password !== requestParams.cpassword) {
+                return { error: config.matchPassword }
+            }
+            const user = await User.findOne({ _id: id })
```

Comment:
Please check, can we use Jwt token here along with Id to find user.

Reply:
we can't use Jwt token here along with Id to find any of the customer
on customer's demand admin can generate new password 
this functionality is for customers and  admin allow to do this.